Rebranding buttons [MAILPOET-3545]

NeosinneR opened the pull request:
[MAILPOET-3545]

[MAILPOET-3545]: https://mailpoet.atlassian.net/browse/MAILPOET-3545

NeosinneR commented:
<img width="1359" alt="emails" src="https://user-images.githubusercontent.com/470616/115553143-5ee46380-a2ad-11eb-923d-ba230165af82.png">
<img width="1358" alt="export" src="https://user-images.githubusercontent.com/470616/115553158-63108100-a2ad-11eb-831a-f27e7be16f1d.png">
<img width="1358" alt="help" src="https://user-images.githubusercontent.com/470616/115553161-64da4480-a2ad-11eb-9f87-2c2f08ac1e2e.png">
<img width="1359" alt="import-1" src="https://user-images.githubusercontent.com/470616/115553165-660b7180-a2ad-11eb-8bb8-10fbecd369e5.png">
<img width="1361" alt="import-2" src="https://user-images.githubusercontent.com/470616/115553173-66a40800-a2ad-11eb-8b9b-bdc427ba8ccf.png">
<img width="1359" alt="key-verification" src="https://user-images.githubusercontent.com/470616/115553177-673c9e80-a2ad-11eb-8606-6755b7aeedb8.png">
<img width="1358" alt="segment" src="https://user-images.githubusercontent.com/470616/115553178-67d53500-a2ad-11eb-8a0f-b9d51af22665.png">
<img width="1359" alt="send-with" src="https://user-images.githubusercontent.com/470616/115553179-686dcb80-a2ad-11eb-8bfe-56831b96c576.png">
<img width="892" alt="storybook-1" src="https://user-images.githubusercontent.com/470616/115553180-69066200-a2ad-11eb-9c89-617207630aff.png">
<img width="890" alt="storybook-2" src="https://user-images.githubusercontent.com/470616/115553182-69066200-a2ad-11eb-81ce-1fc763bb88a9.png">
<img width="1360" alt="subscribers" src="https://user-images.githubusercontent.com/470616/115553185-69066200-a2ad-11eb-9e9b-3552648de9bd.png">

NeosinneR commented:
@j111q I did some changes (fewer primary buttons on a single page) and here are updated screenshots. Let me know if you have more suggestions. 

<img width="1519" alt="Screenshot 2021-05-02 at 16 56 55" src="https://user-images.githubusercontent.com/470616/116817904-f88ef900-ab68-11eb-87a4-fb800b20e744.png">
<img width="1680" alt="Screenshot 2021-05-02 at 16 58 12" src="https://user-images.githubusercontent.com/470616/116817907-fcbb1680-ab68-11eb-9a16-30c392b13021.png">
<img width="1680" alt="Screenshot 2021-05-02 at 16 58 28" src="https://user-images.githubusercontent.com/470616/116817909-ff1d7080-ab68-11eb-9218-9e55bd5a3252.png">
<img width="1680" alt="Screenshot 2021-05-02 at 16 58 36" src="https://user-images.githubusercontent.com/470616/116817910-ffb60700-ab68-11eb-8cf6-5baa14345559.png">
<img width="1675" alt="Screenshot 2021-05-02 at 16 58 49" src="https://user-images.githubusercontent.com/470616/116817915-00e73400-ab69-11eb-8e2a-e6bf30c02db7.png">
<img width="1680" alt="Screenshot 2021-05-02 at 16 59 13" src="https://user-images.githubusercontent.com/470616/116817917-02b0f780-ab69-11eb-800e-19d6ab596172.png">
<img width="1212" alt="Screenshot 2021-05-02 at 17 00 52" src="https://user-images.githubusercontent.com/470616/116817920-05135180-ab69-11eb-950d-e5bfaf8f7d2a.png">
<img width="1297" alt="Screenshot 2021-05-02 at 17 01 05" src="https://user-images.githubusercontent.com/470616/116817922-06447e80-ab69-11eb-8466-96c556bf14ed.png">
<img width="1680" alt="Screenshot 2021-05-02 at 17 01 16" src="https://user-images.githubusercontent.com/470616/116817923-0775ab80-ab69-11eb-8b28-9d7d2f251ff3.png">
<img width="1672" alt="Screenshot 2021-05-02 at 17 01 25" src="https://user-images.githubusercontent.com/470616/116817925-080e4200-ab69-11eb-9601-cdf267232d29.png">
<img width="1680" alt="Screenshot 2021-05-02 at 17 02 12" src="https://user-images.githubusercontent.com/470616/116817926-09d80580-ab69-11eb-8a1f-50962ddadb58.png">
<img width="892" alt="story1" src="https://user-images.githubusercontent.com/470616/116817928-0a709c00-ab69-11eb-85a3-dc29316bb97f.png">
<img width="890" alt="story2" src="https://user-images.githubusercontent.com/470616/116817929-0b093280-ab69-11eb-925f-ed930f4cfc57.png">

j111q commented:
Sorry for the late review here @NeosinneR!

Looking good overall! 👍 A few notes:

**Let's switch always using a `Secondary` button when there's button(s) in the header.**

As you mentioned, those buttons are all equal in importance, but when they're all Primary buttons, they do look very heavy and visually dense. So if we switch them all to secondary buttons, it will work better. Specifically for the screenshots you gave above, I'm referring to the Subscribers, Forms, and Lists pages - all the buttons in the header next to the Mailpoet logo:

![image](https://user-images.githubusercontent.com/65582817/118751329-b4536680-b893-11eb-903c-e8b32c270a12.png)
![image](https://user-images.githubusercontent.com/65582817/118751364-c6cda000-b893-11eb-8945-858a74e5682a.png)
![image](https://user-images.githubusercontent.com/65582817/118751404-d5b45280-b893-11eb-885e-13246011c268.png)


---

**One more thing** - In case it wasn't clear before - the 'blue' should change according to the theme colour of wp-admin [as chosen by the user like](https://www.siteground.com/tutorials/wordpress/color-scheme/) in this screen:

![image](https://user-images.githubusercontent.com/65582817/118752957-9b988000-b896-11eb-8edb-7277e3c7375d.png)

Let me know if you have any other questions or thoughts around these^ comments! Looking great otherwise :) 👍

NeosinneR commented:
@j111q I finally got to fixing those two issues (secondary buttons in header and themed buttons) and here are screenshots :) I'll let a developer review the PR so it can be merged 

<img width="1123" alt="Screenshot 2021-06-02 at 12 06 11" src="https://user-images.githubusercontent.com/470616/120472833-aa0e8d80-c3a6-11eb-8e57-792bf637dc64.png">

<img width="1125" alt="Screenshot 2021-06-02 at 11 16 18" src="https://user-images.githubusercontent.com/470616/120472825-a844ca00-c3a6-11eb-872d-281a2cb9d81c.png">

NeosinneR commented:
@costasovo, there is also a small change in the premium plugin. One shortcoming of this is that in Storybook.js, all button variants look the same because I couldn't import WordPress styles in Storybook.js (they are generated on the fly, and developers can have a different domain where their local WordPress is running).

j111q commented:
Awesome thank you! Looks great! (though I personally hope there are very few users who will choose a purple and green colour palette 😆 )

costasovo commented on assets/css/src/generic/_buttons.scss:
```diff
@@ -1,29 +1,38 @@
+    box-shadow: 0 0 0 1.5px $color-primary-hover;
     color: #fff;
+    outline: none;
     text-decoration: none;
```
Do we need to keep all those styles? After adding WordPress button classes (`button-secondary` etc.) most of these rules are overwritten by CSS from WP. Does it makes sense to keep them?

![Screen Shot on 2021-06-03 at 11-21-48](https://user-images.githubusercontent.com/1082140/120623208-d2a68e00-c45f-11eb-85a6-66fa2688a5ef.png)

costasovo reviewed (changes requested):
Good job! 👍 

I added one question (see comment) + I found one incorrectly styled button on the subscriber's stats page.
![Screen Shot on 2021-06-03 at 11-23-12](https://user-images.githubusercontent.com/1082140/120623637-3335cb00-c460-11eb-967c-e519be68fb59.png)

NeosinneR commented:
@costasovo good catch! Button fixed. 

For the redundant styles - I kept them there on purpose as a fallback. It's possible that they aren't really needed, and `.button-*` classes will always be present in WordPress. But if for any reason they are not, or they are changed in the future, MailPoet buttons would still look usable. 

But I don't have a strong opinion on this, and I'm open to removing them, as it's probably highly unlikely any of that will happen. What do you think?

costasovo commented:
> For the redundant styles - I kept them there on purpose as a fallback. It's possible that they aren't really needed, and .button-* classes will always be present in WordPress. But if for any reason they are not, or they are changed in the future, MailPoet buttons would still look usable.

> But I don't have a strong opinion on this, and I'm open to removing them, as it's probably highly unlikely any of that will happen. What do you think?

I don't have a strong opinion either 😄  I lean slightly towards removing them. I can think of 3 reasons:

1. We want every button to have WordPress's `.button-*` so that they respect admin theme colors. With the fallback styles, a button may look OKish even without the class so a developer may forget to add it.
2. We may by accident increase specificity and those fallback styles may override WP's styles.
3. The existence of those styles makes a developer think about why we have them.

NeosinneR commented:
Fair enough. I removed the styles and also fixed an issue with small buttons. And added a generated CSS, so the Storybook.js buttons look the same as in the plugin.

costasovo reviewed (dismissed):
Thanks!

veljkho reviewed (changes requested):
1. For some reason there are 2 same buttons on Post Notification listing page, [View History](https://d.pr/i/MgSorg) buttons. This should be caused by rebranding buttons.

2. Can we [align Create Segment button](https://d.pr/i/MkrjCO) with filter dropdown on the left?

3. Can we make the `Empty Trash` [button a bit bigger so it fits nicely](https://d.pr/i/QwiNEK) with the left dropdown size? Actually height is not good.

NeosinneR commented:
@veljkho 
1. Fixed
2. Fixed
3. The button is bigger now, but it's still 2px smaller than the inputs on the left. This will be fixed once the rest of the form controls is also restyled, but for now, there will be this 2px difference when the input and button are next to each other.
